### ISSUE TITLE
npm run build and npm run start do not work out of the box, as per readme instructions

### ISSUE THREAD

**Issue by peterschriever** [hyva-themes/magento2-hyva-checkout#103]
When at the `npm run start` or `npm run builld` commands in the readme.md, the following error is displayed instead:

```bash
> tailwind@1.0.0 build
> npx postcss tailwind-source.css -o tailwind-output.css

npx: installed 4 in 1.715s
command not found: postcss
```

reason for this error is that `npm install` for tailwind folder at `view/frontend/web/css/tailwind` are missing. 

After fixing that a long error message regarding a dependency conflict is being shown (again related to #101 and #102 likely):
```
> hyva-checkout@0.1.0 app:start
> react-app-rewired start


There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "eslint": "^6.6.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of eslint was detected higher up in the tree:

  ..redactedPath../magento2/app/code/Hyva/Checkout/reactapp/node_modules/eslint (version: 7.32.0) 

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "eslint" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if ..redactedPath../magento2/app/code/Hyva/Checkout/reactapp/node_modules/eslint is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls eslint in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed eslint.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

P.S. We know this message is long but please read the steps above :-) We hope you find them helpful!
```

Adding `SKIP_PREFLIGHT_CHECK=true` to .env works around this, but isnt a good solution.

**Comment by schmengler** [comment]
Are you using `main` or `develop` branch? The main branch is an outdated prototype, while the develop branch is close to an 1.0 release: https://github.com/hyva-themes/magento2-hyva-checkout/projects/1

**Comment by peterschriever** [comment]
This is using `develop` branch

**Comment by progammer-rkt** [comment]
@schmengler @peterschriever I fixed this issue by updating all packages to the latest and with a single `npm install` command, it will take care of the `tailwindcss` directory npm dependency too. The fix is available in #130

**Comment by peterschriever** [comment]
@programmer-rkt Were currently dealing with a bug in our ci pipeline regarding the preinstall npm command. I pulled from your fork for a new checkout project.

Im on my phone so cant provide the full message currently but it said something along the lines of 'npm cannot run in wd, failed to run lifecycle preinstall'. I havent been able to reproduce locally. Searching around it appears wd stands for working directory and it most likely is regarding the permissions of npm. Perhaps this is somewhat off topic, but i thought id still ask in case u know something about this.

**Comment by progammer-rkt** [comment]
@peterschriever I never came across this issue 🤔 

so the preinstall command is this

```
"preinstall": "node scripts/configurePaymentMethods && npm run tailwind-install",
```
We are doing two operations here. Setting up payment methods and setting up tailwind directory. I think it would be good to check which of these operation fails here. That may give some hints

**Comment by peterschriever** [comment]
It is actually both, it seems the whole lifecycle call of npm run
preinstall fails unfortunately. This then causes the build to also fail to
not having postcss in the tailwind folder. Ill get back to my laptop in
about an hour and provide the error message.

In any case let me know if this requires its own issue.

**Comment by peterschriever** [comment]
Here is the error or rather warning;
```bash
npm WARN lifecycle hyva-checkout@0.1.0~preinstall: cannot run in wd hyva-checkout@0.1.0 node scripts/configurePaymentMethods && npm run tailwind-install (wd=/builds/carnavalskleding/magento2/app/code/Hyva/Checkout/reactapp)
```

which then causes the npm run build to fail;
```bash
$ npm --prefix $CHECKOUT_PATH run build
> hyva-checkout@0.1.0 prebuild /builds/customerRedacted/magento2/app/code/Hyva/Checkout/reactapp
> cd ../view/frontend/web/css/tailwind/ && npm run build-prod
> tailwind@1.0.0 build-prod /builds/customerRedacted/magento2/app/code/Hyva/Checkout/view/frontend/web/css/tailwind
> NODE_ENV=production npm run build && npm run minify && cp ../styles.css ../../../../../reactapp/public/styles.css && npm run output-success
> tailwind@1.0.0 build /builds/customerRedacted/magento2/app/code/Hyva/Checkout/view/frontend/web/css/tailwind
> node_modules/postcss-cli/bin/postcss tailwind-source.css -o tailwind-output.css
sh: 1: node_modules/postcss-cli/bin/postcss: not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! tailwind@1.0.0 build: `node_modules/postcss-cli/bin/postcss tailwind-source.css -o tailwind-output.css`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the tailwind@1.0.0 build script.
```

**Comment by peterschriever** [comment]
I have opened the new ticket #141 to further discuss the issues named above; regarding pipeline issues with npm lifecycle command `preinstall`.

I believe the main problems declared in the starting post have been solved and this can be closed.
